lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -66,10 +66,68 @@ class TraceProcessor {
   }
 
   /**
+   * This method sorts a group of trace events that have the same timestamp. We want to...
+   *
+   * 1. Put E events first, we finish off our existing events before we start new ones.
+   * 2. Order B/X events by their duration, we want parents to start before child events.
+   * 3. If we don't have any of this to go on, just use the position in the original array (stable sort).
+   *
+   * @param {number[]} tsGroupIndices
+   * @param {number[]} timestampSortedIndices
+   * @param {number} indexOfTsGroupIndicesStart
+   * @param {LH.TraceEvent[]} traceEvents
+   * @return {number[]}
+   */
+  static _sortTimestampEventGroup(
+    tsGroupIndices,
+    timestampSortedIndices,
+    indexOfTsGroupIndicesStart,
+    traceEvents
+  ) {
+    const eEventIndices = tsGroupIndices.filter(index => traceEvents[index].ph === 'E');
+    const bxEventIndices = tsGroupIndices.filter(index => traceEvents[index].ph === 'X' ||
+      traceEvents[index].ph === 'B');
+    const otherEventIndices = tsGroupIndices.filter(index => !eEventIndices.includes(index) &&
+      !bxEventIndices.includes(index));
```

Comment:
should probably construct these in a single loop (especially that last `otherEventIndices` :)

Reply:
you think it has a measurable impact? the `n` of these same timestamp groups should always be very small

Reply:
> you think it has a measurable impact? 

maybe not, it is difficult to measure a difference for most traces, true. It may also be more readable (can set a var to `traceEvents[index].ph`), but that's in the eye of the beholder :)